src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -842,9 +842,17 @@ void Compiler::eeDispVar(ICorDebugInfo::NativeVarInfo* var)
     {
         name = "typeCtx";
     }
-    gtDispLclVar(var->varNumber, false);
-    printf("(%10s) : From %08Xh to %08Xh, in ", (VarNameToStr(name) == nullptr) ? "UNKNOWN" : VarNameToStr(name),
-           var->startOffset, var->endOffset);
+    if (0 <= var->varNumber && var->varNumber < lvaCount)
+    {
+        printf("(");
+        gtDispLclVar(var->varNumber, false);
```

Comment:
This is going to lose the formatting of the `%10s`.  I suppose you could use gtGetLclVarName, but then you have to deal with a buffer and size.  I guess you could generalize gtDispLclVar to also have an option to pad to an explicit size (I think the padding might be on the wrong side, but at least the columns would line up).  It might be worth just merging to unblock dumps before thinking about whether or not to do anything.

Reply:
I agree with you on that it won't look great, I just didn't want to block jitdump much longer. I will see if it makes sense to change the buffer to return string directly or do what you are proposing. It is debuggable code so I guess we can afford the simplest return solution.